core/util/src/main/java/org/eclipse/rdf4j/common/net/ParsedIRI.java

```diff
@@ -237,8 +234,7 @@ public static ParsedIRI create(String str) {
 					}
 				}
 			} catch (UnsupportedEncodingException ex) {
-				assert false;
-				return null;
+				throw new IllegalStateException(ex);
```

Comment:
I'm happy to get rid of assertions, but I'd like to dig in a bit deeper before approving this, because it changes the production behavior: previously it would have silently returned here, now it throws an exception.

Reply:
Never mind, has a closer look, totally a "should never happen". Happy to merge.